tracer/src/Datadog.Trace/Debugger/Upload/DiagnosticsUploadApi.cs

```diff
@@ -60,21 +61,23 @@ public override async Task<bool> SendBatchAsync(ArraySegment<byte> data)
 
         private Task<IApiResponse> PostAsync(string uri, ArraySegment<byte> data)
         {
-            if (uri.Contains("diagnostics"))
+            var isLegacy = uri.Contains(LegacyEndpoint);
+            if (isLegacy)
             {
-                var multipart = _apiRequestFactory.Create(new Uri(uri));
-                var multipartRequest = (IMultipartApiRequest)multipart;
-
-                return
-                    multipartRequest
-                         .PostAsync(new MultipartFormItem[]
-                          {
-                              new("event", MimeTypes.Json, "event.json", data)
-                          });
+                return _apiRequestFactory
+                      .Create(new Uri(uri))
+                      .PostAsync(data, MimeTypes.Json);
             }
 
-            var request = _apiRequestFactory.Create(new Uri(uri));
-            return request.PostAsync(data, MimeTypes.Json);
+            var multipart = _apiRequestFactory.Create(new Uri(uri));
+            var multipartRequest = (IMultipartApiRequest)multipart;
```

Comment:
Now I _know_ you're just doing this to cause merge conflicts for me 😝

Reply:
Who? Me? No no, not me 😆 
![image](https://github.com/DataDog/dd-trace-dotnet/assets/19508250/4652615c-fc1d-4777-8b32-ba749717161e)